logging/logs.js

```diff
@@ -0,0 +1,180 @@
+  logging.getEntries(config, function (err, entries) {
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Found %d entries!', entries.length);
+    return callback(null, entries);
+  });
+}
+// [END list_log_entries]
+
+// [START write_log_entry]
+/**
+ * Write a log entry.
+ *
+ * @param {object} options Configuration options.
+ * @param {string} options.name The name of the log to write to.
+ * @param {object} options.resource The resource to associate with the log entry.
+ * @param {string|object} options.entry The body of the log entry.
+ * @param {function} callback The callback function.
+ */
+function writeLogEntry (options, callback) {
+  // Get a reference to an existing log
+  var log = logging.log(options.name);
+
+  // Prepare the entry
+  var entry = log.entry(options.resource, options.entry);
+
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/logging/latest/logging/log
+  log.write(entry, function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Wrote entry to log: %s', options.name);
+    callback(null);
+  });
+}
+// [END write_log_entry]
+
+// [START delete_log]
+/**
+ * Delete a log.
+ *
+ * @param {string} name The name of the log to delete.
+ * @param {function} callback The callback function.
+ */
+function deleteLog (name, callback) {
+  // Get a reference to the Log to be deleted
+  var log = logging.log(name);
+
+  // Delete the log
+  log.delete(function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Deleted log: %s', name);
+    callback(null);
+  });
+}
+// [END delete_log]
+// [END all]
+
+// The command-line program
+var cli = require('yargs');
+
+var program = module.exports = {
+  listLogEntries: listLogEntries,
+  writeLogEntry: writeLogEntry,
+  deleteLog: deleteLog,
+  main: function (args) {
+    // Run the command-line program
+    cli.help().strict().parse(args).argv;
+  }
+};
+
+cli
+  .demand(1)
+  .command('list', 'List log entries in the authenticated project.', {
+    filter: {
+      alias: 'f',
+      type: 'string',
+      requiresArg: true,
+      description: 'Only log entries matching the filter are written.'
+    },
+    limit: {
+      alias: 'l',
+      type: 'number',
+      requiresArg: true,
+      description: 'Maximum number of results to return.'
+    },
+    sort: {
+      alias: 's',
+      type: 'string',
+      requiresArg: true,
+      description: 'Sort results.'
+    }
+  }, function (options) {
+    program.listLogEntries(options, console.log);
+  })
+  .command('write <name> <resource> <entry>', 'Write a log entry.', {}, function (options) {
+    try {
+      options.resource = JSON.parse(options.resource);
+    } catch (err) {
+      console.error('"resource" must be a valid JSON string!');
+    }
+    try {
+      options.entry = JSON.parse(options.entry);
+      program.writeLogEntry(options, console.log);
```

Comment:
Is there an easy way to not throw the JSON string error if `writeLogEntry()` is the one throwing an exception (and not `JSON.parse()`)?